### ISSUE TITLE
Sync does not create schemas in Postgres

### ISSUE THREAD

**Issue by moltar** [typeorm/typeorm#4187]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

1. Specify non-default schema name on the ViewEntity

```ts
@ViewEntity({
  schema: "views",
  expression: (connection: Connection) =>
    connection
      .createQueryBuilder()
      .select('cra.id', 'id') // .....
})
export class PostCategory {
  @ViewColumn()
  public id: number
}
```

```sh
npm run typeorm -- schema:sync
```

```
creating a new view: views.post_category
query: SELECT * FROM current_schema()
query: SELECT * FROM current_schema()
query: CREATE VIEW "views"."post_category" AS SELECT "cra"."id" AS "id" FROM "res" "cra"
query: ROLLBACK
Error during schema synchronization:
{ QueryFailedError: schema "views" does not exist
```

**Comment by Kononnable** [comment]
Same things happen for tables, right?
If I recall correctly this was a design decision. Databases and schema creation is only possible for more privileged db users and such privileges shouldn't be available for applications in production environments.

**Comment by moltar** [comment]
Makes sense. So what is the suggested solution then? Doing manual migrations? Maybe there could be an option to sync schemas. Since it's mostly used and recommended to be used in development, I don't think it has a big security risk. For anything production, it is recommended to do migrations anyways.

**Comment by endurance** [comment]
I think one could take a different stance here and have it be useful to the TypeORM community.

Consider this: the application can run its queries opportunistically, and permissions set by the developers or administrators of the postgres database should manage the allowable scope of said queries.

Postgres has plenty of capabilities to allow or disallow a query from executing depending on user, roles, permissions, etc...

The use case that this design decision now prevents is that when i want to create multiple local databases, I have to come up with a system outside of the library to handle start up. I do not want to track migrations locally for these other databases because the databases only served to mirror systems external to a service.

**Comment by yonatandbinahai** [comment]
Just to be clear, `schema:sync` doesn't work in PostgreSQL?
I tried to run `schema:sync` on an empty PostgreSQL database with some existing entities, didn't get any errors but nothing was created in the DB either.
Any ideas?
